### PR TITLE
Node10 entrypoints use ESM -- Add stubs for /beta, /legacy/, and /internal

### PR DESCRIPTION
**Pull request by DLehenbauer**
Changes:
* Node10 entrypoints now point to ESM (./lib) for dual-format packages.
* Instead of using '/lib/beta' and '/lib/legacy' as export paths, we now emit '/beta.d.ts' and '/legacy.d.ts' in the package root.
* We now additionally emit an './internal.d.ts'.  ATTW now passes the full matrix, with select exclusions of test-only exports.
* Miscellaneous trimming / export map fixes.

### CODE REVIEW

File: azure/packages/azure-service-utils/package.json [hunk 1]
```diff
@@ -44,8 +44,8 @@
 			}
 		}
 	},
-	"main": "./dist/index.js",
-	"types": "./dist/public.d.ts",
+	"main": "lib/index.js",
+	"types": "lib/index.d.ts",
```
Comment (jason-ha) [inline azure/packages/azure-service-utils/package.json:48]: This became untrimmed. (Not going looking for more.)

File: packages/drivers/local-driver/package.json [hunk 2]
```diff
@@ -43,11 +43,11 @@
 			}
 		}
 	},
-	"main": "dist/index.js",
+	"main": "lib/index.js",
 	"browser": {
 		"moniker": "@fluidframework/server-services-client/dist/generateNames.js"
 	},
-	"types": "dist/index.d.ts",
+	"types": "lib/public.d.ts",
```
Comment (DLehenbauer) [inline packages/drivers/local-driver/package.json:50]: Missed trimming?

File: packages/service-clients/azure-client/api-report/azure-client.api.md [hunk 3]
```diff
@@ -68,7 +68,7 @@ export interface AzureContainerVersion {
 
 // @internal @deprecated
 export class AzureFunctionTokenProvider implements ITokenProvider {
-    constructor(azFunctionUrl: string, user?: Pick<AzureMember<any>, "additionalDetails" | "userId" | "userName"> | undefined);
+    constructor(azFunctionUrl: string, user?: Pick<AzureMember<any>, "userId" | "userName" | "additionalDetails"> | undefined);
```
Comment (DLehenbauer) [inline packages/service-clients/azure-client/api-report/azure-client.api.md:71]: enum reordering

File: packages/runtime/runtime-definitions/src/test/types/validateRuntimeDefinitionsPrevious.generated.ts [hunk 4]
```diff
@@ -642,6 +642,18 @@ declare function use_old_InterfaceDeclaration_IGarbageCollectionDetailsBase(
...
+* If breaking change required, add in package.json under typeValidation.broken:
+* "RemovedInterfaceDeclaration_IIdCompressor": {"forwardCompat": false}
+*/
+
+/*
+* Validate back compat by using current type in place of old type
+* If breaking change required, add in package.json under typeValidation.broken:
+* "RemovedInterfaceDeclaration_IIdCompressor": {"backCompat": false}
+*/
+
```
Comment (DLehenbauer) [inline packages/runtime/runtime-definitions/src/test/types/validateRuntimeDefinitionsPrevious.generated.ts:656]: Noted.

File: azure/packages/azure-service-utils/package.json (hunk 1 above)
Comment (DLehenbauer) [reply azure/packages/azure-service-utils/package.json]: Thank you very much for catching that.  I'm embarrassed I didn't see it.

**Review by jason-ha — APPROVED** [review]

**Pull request merged**